src/dune_rules/module_compilation.ml

```diff
@@ -196,49 +211,45 @@ let build_cm cctx ~force_write_cmi ~precompiled_cmi ~cm_kind (m : Module.t)
   in
   let src =
     match phase with
-    | Some Emit -> Path.build linear_fdo
+    | Some Emit ->
+      let linear_fdo =
+        Obj_dir.Module.obj_file obj_dir m ~kind:(Ocaml Cmx)
+          ~ext:Fdo.linear_fdo_ext
+      in
+      Path.build linear_fdo
     | Some Compile | Some All | None -> src
   in
-  let modules = Compilation_context.modules cctx in
+  let opens =
+    let modules = Compilation_context.modules cctx in
+    opens modules m
+  in
   let obj_dirs =
     Obj_dir.all_obj_dirs obj_dir ~mode
     |> List.concat_map ~f:(fun p ->
            [ Command.Args.A "-I"; Path (Path.build p) ])
   in
-  let melange_args =
-    match cm_kind with
-    | Melange Cmj ->
-      let pkg_name_args =
-        match CC.package cctx with
-        | None -> []
-        | Some pkg ->
-          [ Command.Args.A "--bs-package-name"
-          ; A (Package.Name.to_string (Package.name pkg))
-          ]
-      in
-      Command.Args.A "--bs-stop-after-cmj" :: A "--bs-package-output"
-      :: Command.Args.Path (Path.build (CC.dir cctx))
-      :: pkg_name_args
-    | Ocaml (Cmi | Cmo | Cmx) | Melange Cmi -> []
-  in
-  Super_context.add_rule sctx ~dir ?loc:(CC.loc cctx)
+  Super_context.add_rule sctx (* TODO should be [~dir] *)
```

Comment:
What's this about?

Reply:
There will be an upcoming PR to fix this.